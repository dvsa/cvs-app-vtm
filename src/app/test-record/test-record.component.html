<div class="govuk-width-container">
  <a class="govuk-back-link" vtmBackButton>Back</a>

  <main class="govuk-main-wrapper">
    <div class="govuk-grid-row">
      <div class="govuk-grid-column-full">
        <h1 class="govuk-heading-xl">Test record</h1>
      </div>
    </div>

    <form
      [formGroup]="testResultParentForm"
      #testForm="ngForm"
      (ngSubmit)="onSaveTestResult(testForm)"
    >
      <div class="govuk-grid-row">
        <ng-container *ngIf="!editState">
          <div class="govuk-grid-column-one-quarter">
            <button
              class="govuk-button govuk-button--secondary"
              data-module="govuk-button"
              (click)="switchCurrentState('edit')"
            >
              Change test record
            </button>
          </div>
        </ng-container>

        <ng-container *ngIf="editState">
          <div class="govuk-grid-column-one-quarter">
            <button class="govuk-button" data-module="govuk-button" type="submit">
              Save test record
            </button>
            <a class="govuk-link govuk-!-margin-left-2" (click)="switchCurrentState('view')">
              <span>Cancel</span>
            </a>
          </div>
        </ng-container>

        <div class="govuk-grid-column-one-quarter">
          <button class="govuk-button govuk-button--secondary" data-module="govuk-button">
            View certificate
          </button>
        </div>
      </div>

      <div class="govuk-grid-row">
        <div class="govuk-grid-column-full">
          <vtm-accordion>
            <vtm-accordion-item [title]="'Vehicle'">
              <vtm-vehicle
                [testRecord]="testResultObj.testRecord"
                [preparers]="preparers"
                [editState]="editState"
                [isSubmitted]="testForm.submitted"
              ></vtm-vehicle>
            </vtm-accordion-item>
            <vtm-accordion-item [title]="'Test'">
              <vtm-test-section
                [testType]="testResultObj.testType"
                [testRecord]="testResultObj.testRecord"
                [applicableTestTypeIds1]="testTypesApplicable.testSectionApplicable1"
                [applicableTestTypeIds2]="testTypesApplicable.testSectionApplicable2"
              ></vtm-test-section>
            </vtm-accordion-item>
            <span
              [hidden]="
                testTypesApplicable.defectsApplicable?.hasOwnProperty(
                  testResultObj.testType.testTypeId
                )
              "
            >
              <vtm-accordion-item [title]="'Defects'">
                <vtm-defects
                  [testType]="testResultObj.testType"
                  [testRecord]="testResultObj.testRecord"
                ></vtm-defects>
              </vtm-accordion-item>
            </span>
            <span
              [hidden]="
                !testTypesApplicable.seatBeltApplicable?.hasOwnProperty(
                  testResultObj.testType.testTypeId
                ) && !(testResultObj.testRecord.vehicleType == 'psv')
              "
            >
              <vtm-accordion-item [title]="'Seatbelt installation check'">
                <vtm-seatbelt-installation-check
                  [testType]="testResultObj.testType"
                  [editState]="editState"
                  [isSubmitted]="testForm.submitted"
                  [hidden]="
                    !testTypesApplicable.seatBeltApplicable?.hasOwnProperty(
                      testResultObj.testType.testTypeId
                    ) && !(testResultObj.testRecord.vehicleType == 'psv')
                  "
                ></vtm-seatbelt-installation-check>
              </vtm-accordion-item>
            </span>
            <span
              [hidden]="
                !testTypesApplicable.emissionDetailsApplicable?.hasOwnProperty(
                  testResultObj.testType.testTypeId
                ) &&
                (!(testResultObj.testRecord.vehicleType == 'psv') ||
                  !(testResultObj.testRecord.vehicleType.valueOf() == 'hgv'))
              "
            >
              <vtm-accordion-item [title]="'Emission details'">
                <vtm-emission-details
                  [testRecord]="testResultObj.testRecord"
                  [testType]="testResultObj.testType"
                  [editState]="editState"
                  [isSubmitted]="testForm.submitted"
                  [hidden]="
                    !testTypesApplicable.emissionDetailsApplicable?.hasOwnProperty(
                      testResultObj.testType.testTypeId
                    ) &&
                    (!(testResultObj.testRecord.vehicleType == 'psv') ||
                      !(testResultObj.testRecord.vehicleType.valueOf() == 'hgv'))
                  "
                ></vtm-emission-details>
              </vtm-accordion-item>
            </span>
            <vtm-accordion-item [title]="'Visit'">
              <vtm-visit
                [testRecord]="testResultObj.testRecord"
                [editState]="editState"
                [testStations]="testStations"
                [isSubmitted]="testForm.submitted"
              ></vtm-visit>
            </vtm-accordion-item>
            <vtm-accordion-item [title]="'Notes'">
              <vtm-notes
                [testType]="testResultObj.testType"
                [editState]="editState"
                [isSubmitted]="testForm.submitted"
              ></vtm-notes>
            </vtm-accordion-item>
            <span [hidden]="editState">
              <vtm-accordion-item [title]="'Test record history'">
                <vtm-test-history
                  [testRecord]="testResultObj.testRecord"
                  [editState]="editState"
                ></vtm-test-history>
              </vtm-accordion-item>
            </span>
          </vtm-accordion>
        </div>
      </div>
    </form>
  </main>
</div>
