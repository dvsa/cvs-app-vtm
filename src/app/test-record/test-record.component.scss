.govuk-link,
.govuk-link:visited {
  color: #1d70b8 !important;
  font-weight: 500;
  cursor: pointer;
  text-decoration: underline;
  line-height: 2rem;
}

a.govuk-back-link {
  text-decoration: underline;
  cursor: pointer;
}
