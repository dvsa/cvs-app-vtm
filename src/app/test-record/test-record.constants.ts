export const PROHIBITION_RADIO_OPTIONS = { ['Yes']: true, ['No']: false };
