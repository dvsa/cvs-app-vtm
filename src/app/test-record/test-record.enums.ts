export enum EU_VEHICLE_CATEGORY_PSV {
  M1 = 'M1',
  M2 = 'M2',
  M3 = 'M3'
}

export enum EU_VEHICLE_CATEGORY_HGV {
  N1 = 'N1',
  N2 = 'N2',
  N3 = 'N3'
}

export enum EU_VEHICLE_CATEGORY_TRL {
  O1 = 'O1',
  O2 = 'O2',
  O3 = 'O3',
  O4 = 'O4'
}

export enum ODOMETER_READING_UNITS {
  KM = 'kilometres',
  MILES = 'miles'
}

export enum EMISSION_STANDARD {
  EMISSION_STANDARD_1 = '0.10 g/kWh Euro 3 PM',
  EMISSION_STANDARD_2 = '0.03 g/kWh Euro IV PM',
  EMISSION_STANDARD_3 = 'Euro 3',
  EMISSION_STANDARD_4 = 'Euro 4',
  EMISSION_STANDARD_5 = 'Euro 6',
  EMISSION_STANDARD_6 = 'Euro VI',
  EMISSION_STANDARD_7 = 'Full Electric'
}

export enum FUEL_TYPE {
  FUEL_TYPE_1 = 'Diesel',
  FUEL_TYPE_2 = 'Gas-cng',
  FUEL_TYPE_3 = 'Gas-lng',
  FUEL_TYPE_4 = 'Gas-lpg',
  FUEL_TYPE_5 = 'Fuel cell',
  FUEL_TYPE_6 = 'Petrol',
  FUEL_TYPE_7 = 'Full electric'
}

export enum MOD_TYPE {
  MOD_TYPE_1 = 'P - Particulate trap',
  MOD_TYPE_2 = 'M - Modification or change of engine',
  MOD_TYPE_3 = 'G - Gas engine'
}

export enum TEST_STATION_TYPE {
  ATF = 'ATF',
  GVTS = 'GVTS',
  HQ = 'HQ'
}

export enum RESULT {
  FAIL = 'Fail',
  PASS = 'Pass',
  PRS = 'PRS',
  ABANDONED = 'Abandoned'
}

export enum REASON_FOR_ABANDONING_PSV {
  R1 = 'The vehicle was not submitted for test at the appointed time',
  R2 = 'The relevant test fee has not been paid',
  R3 = 'There was no means of identifying the vehicle i.e. the vehicle chassis/serial number was missing or did not relate to the vehicle',
  R4 = 'The registration document or other evidence of the date of first registration was not presented when requested',
  R5 = 'The vehicle was emitting substantial amounts of exhaust smoke so as to make it unreasonable for the test to be carried out',
  R6 = 'The vehicle was in such a dirty or dangerous condition as to make it unreasonable for the test to be carried out',
  R7 = 'The vehicle did not have sufficient fuel and oil to allow the test to be carried out',
  R8 = 'The test could not be completed due to a failure of a part of the vehicle which made movement under its own power impossible',
  R9 = 'Current Health and Safety legislation cannot be met in testing the vehicle',
  R10 = 'The driver and/or presenter of the vehicle declined either to remain in the vehicle or in its vicinity throughout the examination or to drive it or to operate controls or doors or to remove or refit panels after being requested to do so',
  R11 = 'The vehicle exhaust outlet has been modified in such a way as to prevent a metered smoke check being conducted',
  R12 = 'A proper examination cannot be readily carried out as any door engine cover hatch or other access device designed to be opened is locked or otherwise cannot be opened'
}

export enum REASON_FOR_ABANDONING_HGV_TRL {
  R1 = 'The vehicle was not submitted for test at the appointed time',
  R2 = 'The relevant test fee has not been paid',
  R3 = 'The trailer was not accompanied by a suitable motor vehicle',
  R4 = 'There is not permanently fixed to the chassis serial number as shown on the registration document (motor vehicle) or the identification mark issued by the Secretary of State (trailer)',
  R5 = 'A Ministry Plate has been issued and is not fitted to the vehicle or trailer',
  R6 = 'The particulars of the motor vehicle or trailer (e.g. number of axles / axle weights) do not match the VTG6 Ministry Plate fitted to the vehicle',
  R7 = 'The vehicle or motor vehicle by which it is accompanied emits substantial amounts of smoke so as to make it unreasonable for the test to be carried out',
  R8 = 'The vehicle or trailer (any part) was in such a dirty or dangerous condition as to make it unreasonable for the test to be carried out',
  R9 = 'No proof was given that the vehicle used for carrying dangerous/toxic/corrosive or inflammable goods had been cleaned or otherwise rendered safe for test',
  R10 = 'The vehicle did not have sufficient fuel and oil to allow the test to be carried out',
  R11 = 'The vehicle was not loaded as required',
  R12 = 'The test could not be completed due to a failure of a part of the vehicle, or trailer and accompanying motor vehicle which made movement impossible',
  R13 = 'The vehicle was presented for test carrying livestock or other unsuitable material',
  R14 = 'Current Health and Safety legislation cannot be met in testing the vehicle',
  R15 = 'The vehicle exhaust outlet has been modified preventing a metered smoke check',
  R16 = 'The examiner could not open the tachograph',
  R17 = 'The driver and/or presenter of the vehicle refused to or was unable to comply with the instructions of DVSA staff making it impractical or unsafe to continue the test'
}

export enum REASON_FOR_ABANDONING_TIR {
  R1 = 'The vehicle was not submitted for test at the appointed time',
  R2 = 'The relevant test fee has not been paid',
  R3 = 'The trailer was not accompanied by a suitable drawing motor vehicle',
  R4 = 'There is not permanently fixed to the chassis a serial number (motor vehicle) or the identification mark issued by the Secretary of State (trailer)',
  R5 = 'The vehicle or motor vehicle by which it is accompanied emits substantial amounts of smoke so as to make it unreasonable for the test to be carried out',
  R6 = 'The vehicle or trailer (any part) was in such a dirty or dangerous condition as to make it unreasonable for the test to be carried out',
  R7 = 'The vehicle did not have sufficient fuel and oil to allow the test to be carried out',
  R8 = 'The test could not be completed due to a failure of a part of the vehicle, or trailer and accompanying motor vehicle which made movement impossible',
  R9 = 'The vehicle was presented for test carrying unsuitable material',
  R10 = 'Current Health and Safety legislation cannot be met in testing the vehicle',
  R11 = 'The driver and/or presenter of the vehicle refused to or was unable to comply with the instructions of DVSA staff making it impractical or unsafe to continue the test'
}
