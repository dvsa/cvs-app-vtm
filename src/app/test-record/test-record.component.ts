import {
  Component,
  OnInit,
  ChangeDetectionStrategy,
  Input,
  EventEmitter,
  Output
} from '@angular/core';
import { FormBuilder, FormGroup, FormGroupDirective } from '@angular/forms';
import { initAll } from 'govuk-frontend';
import { TestRecordTestType } from '@app/models/test-record-test-type';
import { TestResultModel } from '@app/models/test-result.model';
import { VIEW_STATE } from '@app/app.enums';
import { MatDialog } from '@angular/material/dialog';
import { TestRecordMapper, TestTypesApplicable } from '@app/test-record/test-record.mapper';
import { Preparer } from '@app/models/preparer';
import { DialogBoxComponent } from '@app/shared/dialog-box/dialog-box.component';
import { TestStation } from '@app/models/test-station';
import { PreventLeavePageModalComponent } from '@app/shared/prevent-page-leave-modal/prevent-leave-page-modal.component';

@Component({
  selector: 'vtm-test-record',
  templateUrl: './test-record.component.html',
  changeDetection: ChangeDetectionStrategy.OnPush
})
export class TestRecordComponent implements OnInit {
  @Input() editState: VIEW_STATE;
  @Input() preparers: Preparer[];
  @Input() testResultObj: TestRecordTestType;
  @Input() testStations: TestStation[];
  @Input() testTypesApplicable: TestTypesApplicable;
  @Output() submitTest = new EventEmitter<TestResultModel>();
  @Output() switchState = new EventEmitter<VIEW_STATE>();
  @Output() isFormDirty = new EventEmitter<boolean>();
  testResultParentForm: FormGroup;

  hasDefectsApplicable: boolean;
  hasEmissionApplicable: boolean;
  hasSeatBeltApplicable: boolean;

  constructor(
    private parent: FormGroupDirective,
    protected fb: FormBuilder,
    private dialog: MatDialog,
    private testRecordMapper: TestRecordMapper
  ) {}

  ngOnInit(): void {
    this.hasDefectsApplicable = this.testTypesApplicable.defectsApplicable[
      this.testResultObj.testType.testTypeId
    ];
    this.hasSeatBeltApplicable =
      !this.testTypesApplicable.seatBeltApplicable[this.testResultObj.testType.testTypeId] &&
      !(this.testResultObj.testRecord.vehicleType === 'psv');
    this.hasEmissionApplicable =
      !this.testTypesApplicable.emissionDetailsApplicable[
        this.testResultObj.testType.testTypeId
      ] &&
      (!(this.testResultObj.testRecord.vehicleType === 'psv') ||
        !(this.testResultObj.testRecord.vehicleType.toString() === 'hgv')) &&
      this.testResultObj.testType.testResult !== 'pass';

    initAll();
    this.switchState.emit(VIEW_STATE.VIEW_ONLY);
    this.testResultParentForm = new FormGroup({ testType: new FormGroup({}) });
    this.onFormChanges();
  }

  switchCurrentState(state: string) {
    if (state === 'edit') {
      this.switchState.emit(VIEW_STATE.EDIT);
    } else {
      const dialogRef = this.dialog.open(PreventLeavePageModalComponent, {
        width: '45vw'
      });

      dialogRef.afterClosed().subscribe((result) => {
        if (result) {
          this.switchState.emit(VIEW_STATE.VIEW_ONLY);
          this.testResultObj = JSON.parse(JSON.stringify(this.testResultObj));
          this.testResultParentForm = new FormGroup({ testType: new FormGroup({}) });
        } else {
          this.switchState.emit(VIEW_STATE.EDIT);
        }
      });
    }
  }

  onSaveTestResult(testResultParentForm) {
    const testResultUpdated: TestResultModel = this.testRecordMapper.mapFormValues(
      this.testResultParentForm.getRawValue(),
      this.testResultObj
    );

    const dialogRef = this.dialog.open(DialogBoxComponent, {
      width: '45vw',
      data: { context: 'Enter reason for changing test record', actionName: 'Save test record' }
    });

    dialogRef.afterClosed().subscribe((result) => {
      if (result && result.isSave) {
        testResultUpdated.reasonForCreation = result.data;
        this.submitTest.emit(testResultUpdated);
      }
    });
  }

  onFormChanges(): void {
    this.testResultParentForm.valueChanges.subscribe((val) => {
      this.isFormDirty.emit(true);
    });
  }
}
