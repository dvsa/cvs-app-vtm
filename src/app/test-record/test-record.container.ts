import { Component, OnInit, ChangeDetectionStrategy } from '@angular/core';
import { Observable } from 'rxjs';
import { Store } from '@ngrx/store';
import { IAppState } from '@app/store/state/app.state';
import {
  getTestViewState,
  selectTestType
} from '@app/store/selectors/VehicleTestResult.selectors';
import { ActivatedRoute } from '@angular/router';
import { TestRecordMapper, TestTypesApplicable } from '@app/test-record/test-record.mapper';
import { Preparer } from '@app/models/preparer';
import { TestRecordTestType } from '@app/models/test-record-test-type';
import { VIEW_STATE } from '@app/app.enums';
import { TestResultModel } from '@app/models/test-result.model';
import {
  SetTestViewState,
  UpdateTestResult,
  DownloadCertificate
} from '@app/store/actions/VehicleTestResult.actions';
import { TestStation } from '@app/models/test-station';
import { getPreparers, getTestStations } from '@app/store/selectors/ReferenceData.selectors';
import { TestResultTestTypeNumber } from '@app/models/test-result-test-type-number';

@Component({
  selector: 'vtm-test-record-container',
  template: `
    <ng-container *ngIf="testRecord$ | async as testResultObj">
      <vtm-test-record
        [testResultObj]="testResultObj"
        [preparers]="preparers$ | async"
        [editState]="editState$ | async"
        [testStations]="testStations$ | async"
        [testTypesApplicable]="this.testTypesApplicable"
        (submitTest)="onTestSubmit($event)"
        (switchState)="currentStateHandler($event)"
        (downloadCert)="downloadCertificate($event)"
      >
      </vtm-test-record>
    </ng-container>
  `,
  changeDetection: ChangeDetectionStrategy.OnPush
})
export class TestRecordContainer implements OnInit {
  testRecord$: Observable<TestRecordTestType>;
  preparers$: Observable<Preparer[]>;
  editState$: Observable<VIEW_STATE>;
  testStations$: Observable<TestStation[]>;
  testTypesApplicable: TestTypesApplicable;
  isFormDirty: boolean;
  testTypeNumber: string;

  constructor(
    private store: Store<IAppState>,
    private route: ActivatedRoute,
    private testRecordMapper: TestRecordMapper
  ) {}

  ngOnInit(): void {
    this.testTypesApplicable = {
      seatBeltApplicable: this.testRecordMapper.getTestTypeApplicable('seatBeltApplicable'),
      defectsApplicable: this.testRecordMapper.getTestTypeApplicable('defectsApplicable'),
      emissionDetailsApplicable: this.testRecordMapper.getTestTypeApplicable(
        'emissionDetailsApplicable'
      ),
      anniversaryDateApplicable: this.testRecordMapper.getTestTypeApplicable(
        'anniversaryDateApplicable'
      ),
      expiryDateApplicable: this.testRecordMapper.getTestTypeApplicable('expiryDateApplicable'),
      certificateApplicable: this.testRecordMapper.getTestTypeApplicable('certificateApplicable')
    };

    this.route.paramMap.subscribe((params) => {
      this.testTypeNumber = params.get('id');
    });
    this.testRecord$ = this.store.select(selectTestType);
    this.preparers$ = this.store.select(getPreparers);
    this.editState$ = this.store.select(getTestViewState);
    this.testStations$ = this.store.select(getTestStations);
  }

  onTestSubmit(testResultUpdated: TestResultModel) {
    const testResultTestTypeNumber: TestResultTestTypeNumber = {
      testResultUpdated: testResultUpdated,
      testTypeNumber: this.testTypeNumber,
      testResultsUpdated: {} as TestResultModel[]
    };

    this.store.dispatch(new UpdateTestResult(testResultTestTypeNumber));
  }

  currentStateHandler(state: VIEW_STATE) {
    this.store.dispatch(new SetTestViewState(state));
  }

  downloadCertificate(fileName: string) {
    this.store.dispatch(new DownloadCertificate(fileName));
  }
}
