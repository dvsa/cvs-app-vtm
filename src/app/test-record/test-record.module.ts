import {CommonModule} from '@angular/common';
import {NgModule} from '@angular/core';
import {AuthenticationGuard} from 'microsoft-adal-angular6';
import {SharedModule} from '../shared/shared.module';
import {TestRecordComponent} from '@app/test-record/test-record.component';
import {VehicleComponent} from './vehicle/vehicle.component';
import {TestSectionComponent} from './test-section/test-section.component';
import {DefectsComponent} from './defects/defects.component';
import {SeatbeltInstallationCheckComponent} from './seatbelt-installation-check/seatbelt-installation-check.component';
import {EmissionDetailsComponent} from './emission-details/emission-details.component';
import {VisitComponent} from './visit/visit.component';
import {NotesComponent} from './notes/notes.component';
import {TestHistoryComponent} from './test-history/test-history.component';
import {LibrariesModule} from '@app/shared/libraries/libraries.module';
import {RouterModule} from '@angular/router';

export const COMPONENTS = [
  TestRecordComponent,
  VehicleComponent,
  TestSectionComponent,
  DefectsComponent,
  SeatbeltInstallationCheckComponent,
  EmissionDetailsComponent,
  VisitComponent,
  NotesComponent,
  TestHistoryComponent
];

export const MODULES = [
  CommonModule,
  RouterModule.forChild([
    {path: '', component: TestRecordComponent, canActivate: [AuthenticationGuard], runGuardsAndResolvers: 'always'}
  ]),
  SharedModule,
  LibrariesModule
];

@NgModule({
  imports: [MODULES],
  declarations: [COMPONENTS],
  exports: [COMPONENTS]
})
export class TestRecordModule {
  constructor() {
  }
}
