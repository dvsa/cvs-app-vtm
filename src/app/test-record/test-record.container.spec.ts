import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { RouterTestingModule } from '@angular/router/testing';
import { of } from 'rxjs';
import { CUSTOM_ELEMENTS_SCHEMA } from '@angular/core';
import { INITIAL_STATE, Store } from '@ngrx/store';
import { hot } from 'jasmine-marbles';
import { TestRecordContainer } from '@app/test-record/test-record.container';
import { SharedModule } from '@app/shared/shared.module';
import { ActivatedRoute, convertToParamMap } from '@angular/router';
import { TestRecordMapper, TestTypesApplicable } from '@app/test-record/test-record.mapper';
import { TestRecordTestType } from '@app/models/test-record-test-type';

describe('TestRecordContainer', () => {
  let container: TestRecordContainer;
  let fixture: ComponentFixture<TestRecordContainer>;
  const testRecord = {} as TestRecordTestType;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      imports: [RouterTestingModule, SharedModule],
      providers: [
        TestRecordMapper,
        {
          provide: Store,
          useValue: {
            dispatch: jest.fn(),
            pipe: jest.fn(() => hot('-a', { a: INITIAL_STATE })),
            select: jest.fn()
          }
        },
        {
          provide: ActivatedRoute,
          useValue: {
            queryParams: of({ id: 'W01A34247' }),
            paramMap: of(convertToParamMap({ id: 'W01A34247' }))
          }
        }
      ],
      declarations: [TestRecordContainer],
      schemas: [CUSTOM_ELEMENTS_SCHEMA]
    }).compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(TestRecordContainer);
    container = fixture.componentInstance;
    container.testRecordObservable$ = of(testRecord);
    container.testTypesApplicable = {} as TestTypesApplicable;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(container).toBeTruthy();
  });
});
