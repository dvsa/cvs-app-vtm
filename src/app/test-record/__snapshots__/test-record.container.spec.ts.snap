// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TestRecordContainer should create 1`] = `
<vtm-test-record-container
  editState$={[Function AnonymousSubject]}
  preparers$={[Function AnonymousSubject]}
  route={[Function Object]}
  store={[Function MockStore]}
  testRecord$={[Function BehaviorSubject]}
  testRecordMapper={[Function TestRecordMapper]}
  testStations$={[Function AnonymousSubject]}
  testTypeNumber={[Function String]}
  testTypesApplicable={[Function Object]}
>
  
</vtm-test-record-container>
`;
